File: Dev/unitTest_Engine_cs/Graphics/2D/MapObject2D.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace unitTest_Engine_cs.Graphics._2D
{
    class MapObject2D : EngineTest
    {
        public MapObject2D()
            : base(20)
        {
        }

        protected override void OnStart()
        {
            var scene = new asd.Scene();
            var layer = new asd.Layer2D();
            var obj = new asd.MapObject2D();

            Assert.AreNotEqual<asd.Scene>(scene, null);
            Assert.AreNotEqual<asd.Layer>(layer, null);
            Assert.AreNotEqual<asd.MapObject2D>(obj, null);

            asd.Engine.ChangeScene(scene);
            scene.AddLayer(layer);
            layer.AddObject(obj);

            var g = asd.Engine.Graphics;
            var texture = g.CreateTexture2D("Data/Map/MapTest.png");

            var chip1 = new asd.Chip2D();
            chip1.Texture = texture;
            chip1.Position = new asd.Vector2DF(0, 0);
            obj.AddChip(chip1);

            var chip2 = new asd.Chip2D();
            chip2.Texture = texture;
            chip2.Position = new asd.Vector2DF(300, 0);
            obj.AddChip(chip2);

            obj.Position = new asd.Vector2DF(0, 0);
        }

        protected override void OnUpdating()
        {

        }
    }
}

File: Dev/unitTest_Engine_cpp_gtest/Graphics/2D/MapObject2D.cpp
#include "../../EngineTest.h"
#include <Altseed.h>
#include <gtest/gtest.h>

using namespace std;
using namespace asd;

class Graphics_MapObject2D : public EngineTest
{
public:
	Graphics_MapObject2D(bool isOpenGLMode) : EngineTest(u"MapObject2D", isOpenGLMode, 30) {}

protected:
	void OnStart()
	{
		auto scene = make_shared<Scene>();
		auto layer = make_shared<Layer2D>();
		auto object = make_shared<MapObject2D>();

		ASSERT_NE(scene, nullptr);
		ASSERT_NE(layer, nullptr);
		ASSERT_NE(object, nullptr);

		asd::Engine::ChangeScene(scene);
		scene->AddLayer(layer);
		layer->AddObject(object);

		auto g = asd::Engine::GetGraphics();
		auto texture = g->CreateTexture2D(u"Data/Map/MapTest.png");
		auto chip1 = std::make_shared<Chip2D>();
		chip1->SetTexture(texture);
		chip1->SetPosition(Vector2DF(0, 0));
		object->AddChip(chip1);

		auto chip2 = std::make_shared<Chip2D>();
		chip2->SetTexture(texture);
		chip2->SetPosition(Vector2DF(300, 0));
		object->AddChip(chip2);
	}
};

ENGINE_TEST(Graphics, MapObject2D)